src/ScheduleFunctions.cpp

```diff
@@ -2249,6 +2249,10 @@ bool validate_schedule(Function f, const Stmt &s, const Target &target, bool is_
         return true;
     }
 
+    if (f.schedule().ring_buffer().defined() && store_at == hoist_storage_at) {
+        user_error << "Func \"" << f.name() << "\" is scheduled ring_buffer(), but has matching store_at and hoist_storage levels.\n";
```

Comment:
scheduled with ring_buffer()

Reply:
"but has matching store_at and hoist_storage levels." -- so is the fix to ensure that these levels don't match? Let's tell them how to fix the error if possible.